tls/s2n_x509_validator.c

```diff
@@ -463,13 +463,15 @@ s2n_cert_validation_code s2n_x509_validator_validate_cert_stapled_ocsp_response(
     }
 
     /* Important: this checks that the stapled ocsp response CAN be verified, not that it has been verified. */
-    const int ocsp_verify_err = OCSP_basic_verify(basic_response, cert_chain, validator->trust_store->trust_store, 0);
-    /* do the crypto checks on the response.*/
-    if (!ocsp_verify_err) {
+    const int ocsp_verify_res = OCSP_basic_verify(basic_response, cert_chain, validator->trust_store->trust_store, 0);
+
+    /* OCSP_basic_verify() returns 1 on success, 0 on error, or -1 on fatal error such as malloc failure. */
+    if (ocsp_verify_res != _OSSL_SUCCESS) {
         ret_val = S2N_CERT_ERR_UNTRUSTED;
```

Comment:
I think your callout is probably right: `S2N_CERT_ERR_UNTRUSTED` doesn't seem like the correct error. I would suggest either `S2N_ERR_SAFETY` for a generic internal error, or if we think it'll usually be for malloc failures we could do `S2N_ERR_ALLOC`.

I think in the end we at least want the error type to be `S2N_ERR_T_INTERNAL` rather than `S2N_ERR_T_PROTO`.

Reply:
I think `S2N_ERR_ALLOC` would make sense, given the documentation calls that out

Reply:
I mean, we do need to make sure that if OCSP_basic_verify() = 0, we don't treat it as an S2N_ERR_SAFETY, right?

Reply:
Yeah. But actually this function returns `s2n_cert_validation_code`, so I think we may be limited to those error codes: https://github.com/aws/s2n-tls/blob/c175de5c02bf51920d8b4a10959d5b4a0b0a31b9/tls/s2n_x509_validator.h#L32-L40

Reply:
Ah you're totally right. We should probably add a new code then.

Reply:
I've added `S2N_CERT_ERR_INTERNAL_ERROR`